src/App.tsx

```diff
@@ -1,8 +1,116 @@
+import { useState } from 'react';
+import { TextField, Box, Button } from '@mui/material';
 import './App.css';
 
 const App: React.FC = () => {
+  const [digits, setDigits] = useState<string[]>(new Array(8).fill(''));
+  const [lastDigit, setLastDigit] = useState<number | null>(null); 
+
+  const baseUrl: string = "http://localhost:3000";
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>, index: number) => {
+    const newDigits = [...digits];
+    const input = event.target.value;
+
+    if (input.match(/^\d$/)) {
+      newDigits[index] = input;
+      setDigits(newDigits);
+
+      if (input && index < digits.length - 1) {
+        const nextInput = document.getElementById(`digit-${index + 1}`) as HTMLInputElement;
+        if (nextInput) {
+          nextInput.focus();
+        }
+      }
+    } else {
+      event.preventDefault();
+    }
+  };
+
+  const handlePaste = (event: React.ClipboardEvent<HTMLInputElement>) => {
+    const pastedData = event.clipboardData.getData('Text').split('');
+    const newDigits = [...digits];
```

Comment:
why u need [...digits] , seems like empty array ca work, you are overriding it anyway